Makefile

```diff
@@ -1,35 +1,44 @@
 BUILD_CHANNEL?=local
-TOOL_BIN = bin/gotools/$(shell uname -s)-$(shell uname -m)
-PATH_WITH_TOOLS="`pwd`/$(TOOL_BIN):$(HOME)/go/bin/:${PATH}"
-GIT_REVISION = $(shell git rev-parse HEAD | tr -d '\n')
-TAG_VERSION?=$(shell git tag --points-at | sort -Vr | head -n1)
-GO_BUILD_LDFLAGS = -ldflags "-X 'main.Version=${TAG_VERSION}' -X 'main.GitRevision=${GIT_REVISION}'"
+TOOL_BIN := $(shell pwd)/bin/tools/$(shell uname -s)-$(shell uname -m)
+GIT_REVISION := $(shell git rev-parse HEAD | tr -d '\n')
+TAG_VERSION ?= $(shell git tag --points-at | sort -Vr | head -n1)
+GO_BUILD_LDFLAGS := -ldflags "-X 'main.Version=${TAG_VERSION}' -X 'main.GitRevision=${GIT_REVISION}'"
+SHELL := /usr/bin/env bash
+export PATH := $(TOOL_BIN):$(PATH)
+export GOBIN := $(TOOL_BIN)
+
+ifneq (, $(shell which brew))
+	EXTRA_CMAKE_FLAGS := -DCMAKE_PREFIX_PATH=$(shell brew --prefix) -DQt5_DIR=$(shell brew --prefix qt5)/lib/cmake/Qt5
+	export PKG_CONFIG_PATH := $(shell brew --prefix openssl@3)/lib/pkgconfig
+endif
+
+default: build
 
-ARTIFACT="~/go/bin/artifact"
+artifact-pull: $(TOOL_BIN)/artifact
+	artifact pull
 
-artifact-pull:
-	PATH=${PATH_WITH_TOOLS} artifact pull
+$(TOOL_BIN)/artifact:
+	 go install go.viam.com/utils/artifact/cmd/artifact
 
-bufinstall:
-	sudo apt-get install -y protobuf-compiler-grpc libgrpc-dev libgrpc++-dev || brew install grpc openssl --quiet
+$(TOOL_BIN)/buf:
+	go install github.com/bufbuild/buf/cmd/buf@v1.8.0
 
-bufsetup:
-	GOBIN=`pwd`/grpc/bin go install github.com/bufbuild/buf/cmd/buf@v1.8.0
-	ln -sf `which grpc_cpp_plugin` grpc/bin/protoc-gen-grpc-cpp
+$(TOOL_BIN)/protoc-gen-grpc-cpp:
```

Comment:
[q] what is the benefit of moving to this more complex make target name?

Reply:
gotcha, so thats why other repos were updated to use `dist/buf` and similar setups

Reply:
Yeah, though in that case, it was using a slightly different pattern where the "dist" folder contains touchfiles (just empty files with specific names) to basically flag when a job had been "done." You can then reset everything by just wiping those files out, causing a clean build to happen. Sometimes when you're linking in more complex tooling (that might have dozens of things it generates, like the buf generate command does) that's an easier way than setting up targets for EVERY output file. Especially if the output files often change or are variable in number.